redis/asyncio/connection.py

```diff
@@ -1519,36 +1532,34 @@ class BlockingConnectionPool(ConnectionPool):
         >>> pool = BlockingConnectionPool(timeout=5)
     """
 
+    __slots__ = (
+        "queue_class",
+        "timeout",
+        "pool",
+    )
+
     def __init__(
         self,
         max_connections: int = 50,
         timeout: Optional[int] = 20,
         connection_class: Type[Connection] = Connection,
-        queue_class: Type[asyncio.Queue] = asyncio.LifoQueue,
```

Comment:
The LifoQueue ensures that _hot_ connections are the ones re-used first, which is probably beneficial.

Reply:
There was no difference in my simple tests, maybe i did something wrong but Lifo had 0 impact

Reply:
Well, I´m not sure there will be a difference, or that this was the original intent of the LifoQueue.  But LIFO is often used in caching, to maintain _hot_ information.  There is a degenerate case which can happen with Fifo, which is that if you pull always the oldest connections, you may end up with all the connections having timed out, or been disconnected due to idleness or other such things.   But maybe that doesn't matter at all.   Maybe the only intention of the LifoQueue was to keep those `None` objects at the end.